python/cudf/cudf/core/frame.py

```diff
@@ -573,6 +574,40 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
+    def _explode(self, explode_column: Any, ignore_index: bool):
+        """Helper function for `explode` in Series and Dataframe.
+        if the designated column to explode is non-nested, a copy
+        of the frame is returned. Otherwise, if ignore_index is
+        set, the original index is not exploded and will use
+        a `RangeIndex`.
+        """
+        if (
+            isinstance(self, (cudf.Series, cudf.DataFrame))
```

Comment:
This is a circular reference. `Series` and `DataFrame` inherit from `Frame` where we shouldn't need to use them in this implementation.

Reply:
The intention was to circumvent mypy's error message to access the `._index` field and `._from_data` method. It's better to move the usage to one level up.